qutip/solver/heom/bofin_baths.py

```diff
@@ -412,8 +594,9 @@ def terminator(self):
         delta, L = self._dl_terminator.terminator(self.exponents)
         return delta, L
 
-    def _matsubara_params(self, lam, gamma, T, Nk):
-        """ Calculate the Matsubara coefficents and frequencies. """
+    @staticmethod
```

Comment:
It's a bad idea to make this a staticmethod unless it's called from outside the class because it limits refactorying opportunties. E.g. If this method wants to call other methods, it now has to do `ClassName._other_method(...)` instead of `self._other_method`.

Making it a classmethod is also acceptable.

Reply:
It is called from outside the class. However, it would also work as a classmethod.

Reply:
I should probably have not done reviewing so late at night. :/

Reply:
Oh my previous comment wasn't quite correct either though, sorry. This function here does not get called from outside, but its counterpart in the underdamped bath class does. Might be best to make both into classmethods, then?